src/compiler/binder.ts

```diff
@@ -3941,28 +3951,27 @@ export function getContainerFlags(node: Node): ContainerFlags {
             // falls through
         case SyntaxKind.Constructor:
         case SyntaxKind.FunctionDeclaration:
+        case SyntaxKind.ClassStaticBlockDeclaration:
+            return ContainerFlags.IsContainer | ContainerFlags.IsControlFlowContainer | ContainerFlags.HasLocals | ContainerFlags.IsFunctionLike;
         case SyntaxKind.MethodSignature:
         case SyntaxKind.CallSignature:
         case SyntaxKind.JSDocSignature:
         case SyntaxKind.JSDocFunctionType:
         case SyntaxKind.FunctionType:
         case SyntaxKind.ConstructSignature:
         case SyntaxKind.ConstructorType:

```

Comment:
Type-level AST nodes had `ContainerFlags.IsControlFlowContainer` here. This was messing up some of the changes I made since it was interfering with the implemented `seenThisKeyword` tracking. I don't see why those would be considered control flow containers and there are no tests proving it was needed.

Other changes in this function are basically of the same kind - I just removed `ContainerFlags.IsControlFlowContainer` from the type-level nodes.